### ISSUE TITLE
[FEATURE] Implement a more type-aware variant of inlining

### ISSUE THREAD

**Issue by Kukovec** [informalsystems/apalache#1108]
A followup to #1107, it has become clear that `InlinerOfUserOper` would benefit from a redesign, to account for the need for type calculus.

**Comment by konnov** [comment]
I think we have fixed the related issues, right?

**Comment by Kukovec** [comment]
Yes, this was closed by https://github.com/informalsystems/apalache/pull/1217
